lstests/core/helper.py

```diff
@@ -0,0 +1,148 @@
+#! /usr/bin/env python3
+
+# standard libraries
+import copy
+import json
+import subprocess
+import time
+import typing
+# user-defined modules
+if __package__:
+    from . import color
+    from . import logger
+else:
+    import color
+    import logger
+
+# --------------- utility ---------------------------------------------------------------------------------------------
+
+def override(*args):
+    """
+    Example
+    -------
+    >>> a, b, c = 1, 2, None
+    >>> override(a, b, c)
+    2
+    >>> default_value = 1
+    >>> value = 2
+    >>> command_line_value = 3
+    >>> value = override(default_value, value, command_line_value)
+    >>> value
+    3
+    """
+    for x in reversed(args):
+        if x is not None:
+            return x
+    return None
+
+# --------------- format ----------------------------------------------------------------------------------------------
+
+def get_time(date=True, precision=3, local_time=False, time_zone=False):
+    now = time.time()
+    fmt = "%H:%M:%S"
+    if date:
+        fmt = "T".join(["%Y-%m-%d", fmt])
+    if precision:
+        subsec = int((now - int(now)) * (10 ** precision))
+        fmt = ".".join([fmt, f"{subsec:0{precision}d}"])
+    if time_zone:
+        fmt = " ".join([fmt, "%Z"])
+    convert = time.localtime if local_time else time.gmtime
+    return time.strftime(fmt, convert(now))
+
+
+def make_tokens(amount: int, ndigits=4, symbol="SYS"):
+    """
+    Doctest
+    -------
+    >>> print(make_tokens(37.5e6))
+    37500000.0000 SYS
+    >>> print(make_tokens(1, 3, 'ABC'))
+    1.000 ABC
+    """
+    return "{{:.{}f}} {{}}".format(ndigits).format(amount, symbol)
+
+# --------------- string ----------------------------------------------------------------------------------------------
+
+def plural(word, count, suffix="s"):
+    if isinstance(word, tuple) or isinstance(word, list):
+        return word[0] if count <= 1 else word[1]
+    else:
+        return word if count <= 1 else  word + suffix
+
+
+def singular(word, count, suffix="s"):
+    if isinstance(word, tuple) or isinstance(word, list):
+        return word[0] if count <= 1 else word[1]
+    else:
+        return word  + suffix if count <= 1 else  word
+
+
+def pad(text, total, left=0, char=" ", sep="", textlen=None) -> str:
+    """textlen is a hint for visable length of text"""
+    textlen = vislen(text) if textlen is None else textlen
+    offset = len(text) - textlen
+    return (char * left + sep + text + sep).ljust(total + offset, char)
+
+
+def vislen(s: str):
+    """Return visible length of a string after removing color codes in it."""
+    return len(color.REGEX.sub("", s))
+
+
+def squeeze(s, maxlen, tail=0):
```

Comment:
Consider using `shorten` from the `textwrap` module, available since Python 3.4.  All supported build platforms are using a Python 3 version newer than 3.4.

Reply:
Thank you for that suggestion. Personally I would like to keep the `squeeze` function, as it allows a customizable tail, so it's possible to tell the difference after you squeeze, say, strings `"catchup-info-1"` and `"catchup-info-2"`. The former becomes `"catch..-1"`. The latter becomes `"catch..-2"`.